tests/main/conftest.py

```diff
@@ -86,6 +89,8 @@ def create_mock_segy_rev1(  # noqa: PLR0913
     rev1_spec = registry.get_spec(SegyStandard.REV1)
     rev1_spec.trace.data_descriptor.samples = num_samples
 
+    rev1_spec.endianness = Endianness.LITTLE
```

Comment:
Does this update the other components of the spec to all use little endian or is this the only part of the spec where it is checked at?

Reply:
this updates the whole spec (see the new model_validate methods on the pydantic models); i.e. makes everything consistent. The reason I kept the sub-model endianness in structfielddtype is because there are some places we don't have acccess to the full spec and need to get it from a sub-model (like binary header). It's usually used during data type models to numpy dtype. Like when you do binary_spec.dtype it has to know the endianness, and if we don't have the full spec in scope we can't build the correct dtype.

I'll think and see if there's a way to make the dtype methods work some other way but I'm not very optimistic because then the dtype needs to be parametrized or something like that.

Reply:
Okay that makes sense to me.